test/spec/services/crud-service.js

```diff
@@ -0,0 +1,124 @@
+'use strict';
+
+var crudService;
+var httpBackend;
+var url = 'http://locahost/user';
+var mockUsers = [
+  {
+    name: 'Max'
+  },
+  {
+    name: 'Devin'
+  }
+];
+
+describe('The Crud Service', function() {
+
+  beforeEach(module('adminApp'));
+
+  beforeEach(inject(function($injector) {
+    crudService = $injector.get('crudService');
+    httpBackend = $injector.get('$httpBackend');
+  }));
+
+  describe('create() method', function() {
+    it('should make a POST request and create a record', function() {
+      var options = {
+        url: url
+      };
+      var payload = {
+        name: 'Max'
+      };
+      httpBackend.expectPOST(/user/).respond(201,mockUsers[0]);
+      crudService.create(payload,options).then(function(newUser){
+        expect(newUser).toEqual(payload);
+        httpBackend.flush();
+      });
+    });
+  });
+
+
+  describe('retrieve() method', function() {
+
+    it('should make a GET request and return a single record', function() {
+      var options = {
+        url: url
+      };
+      httpBackend.expectGET(/user\/\d+$/).respond(200,mockUsers[0]);
+      crudService.retrieve(1,options).then(function(user){
+        expect(user).toEqual(mockUsers[0]);
+        httpBackend.flush();
+      });
+    });
+
+    it('should be able to cache a call', function() {
```

Comment:
Good questions. This is not meant to test if a call returns cached results or not, but rather pass the _cache_ parameter to the `$resource` / `$http services` that exist under the `crudService`. The caching part is already tested in the `$http` service but I'll update the test to be more descriptive.